src/hotspot/share/memory/metaspace.hpp

```diff
@@ -234,212 +150,115 @@ class Metaspace : public AllStatic {
     return mdType == ClassType && using_class_space();
   }
 
-  static bool initialized() { return _initialized; }
+  static bool initialized();
 
 };
 
-// Manages the metaspace portion belonging to a class loader
+// ClassLoaderMetaspace is an inbetween-object between a CLD and its MetaspaceArena(s).
+//
+// A CLD owns one MetaspaceArena if compressed class space is off, two if its one
+// (one for allocations of Klass* structures from class space, one for the rest from
+//  non-class space).
+//
+// ClassLoaderMetaspace only exists to hide this logic from upper layers:
+//
```

Comment:
I would suggest rewriting the comments to

```
// A ClassLoaderMetaspace manages  MetaspaceArena(s) for a CLD.
//
// A CLD owns one MetaspaceArena if UseCompressedClassPointers is false. Otherwise
// it owns two -- one for the Klass* objects from the class space, one for the other types
// of MetaspaceObjs from the non-class space.
```

(I think "hide this logic ...." can be omitted. We have lots of abstractions so there's no need to explicitly call it out).

Reply:
Okay, sounds good.